packages/components/file-uploader/src/file-uploader.css

```diff
@@ -0,0 +1,421 @@
+  border-radius: var(--ts-unit);
+  color: var(--ts-color-white);
+  cursor: pointer;
+  font-size: 12px;
+  letter-spacing: 0.5px;
+  padding: calc(0.2 * var(--ts-unit-half)) var(--ts-unit-half) 0;
+  text-transform: uppercase;
+}
+
+:host > .file-upload-wrapper > label {
+  align-items: center;
+  color: var(--ts-color-gray-darker);
+  display: flex;
+  font-size: calc(1px + var(--ts-unit-half));
+  padding-left: calc(1px + var(--ts-unit-half));
+}
+
+:host > .file-upload-wrapper > label > svg {
+  height: calc(1.5 * var(--ts-unit-half));
+  margin: calc(-0.2 * var(--ts-unit-half)) calc(0.5 * var(--ts-unit-half)) 0 0;
+  width: calc(1.5 * var(--ts-unit-half));
+}
+
+/* Uploading card shared styles....................................... */
+
+:host > .file-card-wrapper > .file-card {
+  margin-top: calc(2 * var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card > .file-info--wrapper {
+  align-items: center;
+  display: flex;
+}
+
+:host > .file-card-wrapper > .file-card > .file-info--wrapper > .icon-wrapper {
+  margin-right: var(--ts-unit-half);
+  width: calc(3.5 * var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card--failed label,
+:host > .file-card-wrapper > .file-card--download label {
+  font-weight: 600;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+:host > .file-card-wrapper > .file-card--failed .file-upload--info,
+:host > .file-card-wrapper > .file-card--download > .file-upload--info > div {
+  line-height: 14px;
+}
+
+:host > .file-card-wrapper > .file-card--failed small,
+:host > .file-card-wrapper > .file-card--download small {  
+  font-size: calc(1px + var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card--failed small:first-of-type,
+:host > .file-card-wrapper > .file-card--download small {
+  color: #7F7F7F;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+/* Uploading state................................................... */
+
+:host > .file-card-wrapper > .file-card {
+  cursor: default;
+  height: calc(6.5 * var(--ts-unit-half));
+  margin-bottom: 0;
+}
+
+:host > .file-card-wrapper > .file-card:hover {
+  border: var(--ts-border);
+}
+
+:host > .file-card-wrapper .icon { 
+  align-items: center;
+  border-radius: 50%;
+  color: var(--ts-color-white);
+  display: flex;
+  justify-content: center;
+}
+
+:host > .file-card-wrapper > .file-card .icon:first-of-type {
+  background-color: var(--ts-color-blue);
+  height: 28px;
+  width: 28px;
+}
+
+:host > .file-card-wrapper .icon:first-of-type > span {
+  font-size: var(--ts-unit);
+  font-weight: bold;
+}
+
+:host > .file-card-wrapper .file-upload--info,
+:host > .file-card--download .file-upload--info {
+  display: flex;
+  flex-direction: column;
+  max-width: calc(16.5 * var(--ts-unit-half));
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+:host > .file-card-wrapper > .file-card--uploading .progress {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Simplify this over-specified selector. Maximum allowed depth: 3. Actual depth: 4 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aoverspecific-selectors)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this usage of the experimental "host" pseudo. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aexperimental-pseudo-usage)